### ISSUE TITLE
PRINT statement in trigger results in error and promise rejection

### ISSUE THREAD

**Issue by YardGnomeNinja** [TimelordUK/node-sqlserver-v8#145]
I discovered this issue while using the `msnodesqlv8` driver via [tediousjs/node-mssql](https://github.com/tediousjs/node-mssql).

I think the problem occurs in the function `nativeNextResult(queryId)` starting on line 56 of `msnodesqlv8/lib/reader.js`. 

When `native.nextResult(queryId, (e, res)` is called on a query that fires a trigger containing a PRINT statement, an error is thrown that causes the promise to be rejected.

Here is the error I received during my testing:
```
e: [
  0: {
    code: 0,
    message: "[Microsoft][SQL Server Native Client 11.0][SQL Server]CustomerUpdated fired",
    sqlstate: "01000",
    stack: "Error: [Microsoft][SQL Server Native Client 11.0][SQL Server]CustomerUpdated fired"
  }
]
```
I would be willing to submit a PR to attempt to resolve this, but I'm unsure what would be the best course of action to avoid breaking expected functionality.

I originally submitted a bug to [tediousjs/node-mssql](https://github.com/tediousjs/node-mssql/issues/1004) regarding this issue, but I traced it to this location.

**Comment by YardGnomeNinja** [comment]
I dug into this further in an attempt to identify the best way to resolve the issue. I was not as successful as I would have liked, but I did find that by changing `reject(e)` on line 61 of `msnodesqlv8/lib/reader.js` to `resolve(e)` the hang would no longer occur.

I feel like this may not be the correct way to fix this, however.

I tried to determine why the driver is considering the output text as an error, but was unsuccessful in pinpointing it so far.

**Comment by TimelordUK** [comment]
so using the raw library and writing this test I see following.  That is the print is routed as an info message and no error is raised.  I would need an exact test that reproduces what you are seeing. The idea was to route print via a different event other than errors.

```javascript
    test('use print statement capture print', done => {
    const q = theConnection.query('print \'hello world!\'; select 1 as one')
    const errors = []
    const info = []
    const rows = []
    let currentRow = []
    let metadata = null
    let lastColumn = 0

    const expectedInfo = [
      {
        sqlstate: '01000',
        code: 0,
        message: '[Microsoft][SQL Server Native Client 11.0][SQL Server]hello world!'
      }
    ]
    q.on('error', e => {
      errors.push(e)
    })

    q.on('info', m => {
      info.push({
        sqlstate: m.sqlstate,
        code: m.code,
        message: m.message
      })
    })

    q.on('done', () => {
      assert.deepStrictEqual(expectedInfo, info)
      assert(errors.length === 0)
      assert.deepStrictEqual(1, rows.length)
      assert.deepStrictEqual(rows, [
        [
          1
        ]
      ])
      done()
    })

    q.on('meta', (meta) => {
      metadata = meta
      currentRow = [metadata.length]
      lastColumn = metadata.length - 1
    })

    q.on('column', (index, data) => {
      currentRow[index] = data
      if (index === lastColumn) {
        rows.push(currentRow)
        currentRow = [metadata.length]
      }
    })
  })

````

**Comment by YardGnomeNinja** [comment]
[The project I'm working on](https://github.com/YardGnomeNinja/json-to-sql) has everything necessary to run the test with fairly minimal effort. 

To set it up:
- run the db scripts in `test/db/create` on a new database
- edit `test/fixtures/initialize.json` to use your database with a trusted connection (or edit 'src/sql-runner.translateConfig()' to force it to use `msnodesqlv8`)
- run 'json-to-sql' with 'test/fixtures/initialize.json'

If you have VS Code you can use the Launch Profile profile with the debugger and put a breakpoint at `/node_modules/msnodesqlv8/lib/reader.js' line 61 and step in from that point.

It will demonstrate the problem in full when it tries to update a customer record.

**Comment by TimelordUK** [comment]
OK will check out your project and see what is going on

**Comment by TimelordUK** [comment]
ok i checked out the code

I simply do not see the issue - I have a breakpoint on 61 does not hit.  I have breakpoint on line 58 I see many hits.

```javascript
   function nativeNextResult (queryId) {
      return new Promise((resolve, reject) => {
        native.nextResult(queryId, (e, res) => {
          setImmediate(() => {
            if (e) {
              reject(e)
            } else {
              resolve(res)
            }
          })
        })
      })
    }
```

here is my output


```console
"C:\Program Files\nodejs\node.exe" C:\Users\Stephen\dev\js\json-to-sql\dist\app.js ../test/fixtures/initialize.json
Debugger listening on ws://127.0.0.1:57058/0a4a0289-9428-40a6-abd7-2be7f2810bab
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
Opening connection to '(localdb)\node' database 'scratch'...
Opening connection complete: 00:00:00.0.29.
Executing JSON fixture '../test/fixtures/initialize.json'...
!!!Mutations disabled!!!
-----
table 'customer'
-----
SELECT IIF (EXISTS (SELECT 1 FROM sys.Objects WHERE object_id = object_id(N'customer') AND type IN (N'U',N'V')), 'true', 'false') as 'object_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { object_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
SELECT IIF (EXISTS (SELECT 1 FROM customer WHERE customerId=1), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'false' } ],
  output: {},
  rowsAffected: []
}
INSERT INTO customer (first, last) VALUES ('Mike', 'Wazowski')
!!Not Executed!!
-----
SELECT IIF (EXISTS (SELECT 1 FROM customer WHERE customerId=2), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'false' } ],
  output: {},
  rowsAffected: []
}
INSERT INTO customer (first, last) VALUES ('James', 'Sullivan')
!!Not Executed!!
-----
SELECT IIF (EXISTS (SELECT 1 FROM customer WHERE customerId=3), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'false' } ],
  output: {},
  rowsAffected: []
}
INSERT INTO customer (first, last) VALUES ('Boo', 'The Kid')
!!Not Executed!!
-----
-----
table 'product'
-----
SELECT IIF (EXISTS (SELECT 1 FROM sys.Objects WHERE object_id = object_id(N'product') AND type IN (N'U',N'V')), 'true', 'false') as 'object_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { object_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
SELECT IIF (EXISTS (SELECT 1 FROM product WHERE productId=1), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'false' } ],
  output: {},
  rowsAffected: []
}
INSERT INTO product (name, description, priceUsd) VALUES ('Spatula', 'flip flip flip', 5)
!!Not Executed!!
-----
SELECT IIF (EXISTS (SELECT 1 FROM product WHERE productId=2), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'false' } ],
  output: {},
  rowsAffected: []
}
INSERT INTO product (name, description, priceUsd) VALUES ('Hat', null, 50)
!!Not Executed!!
-----
-----
table '[order]'
-----
SELECT IIF (EXISTS (SELECT 1 FROM sys.Objects WHERE object_id = object_id(N'[order]') AND type IN (N'U',N'V')), 'true', 'false') as 'object_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { object_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
SELECT IIF (EXISTS (SELECT 1 FROM [order] WHERE orderId=1), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'false' } ],
  output: {},
  rowsAffected: []
}
INSERT INTO [order] (customerId, totalUsd, createdOn) VALUES (1, 15, '2020-1-1 14:00:00')
!!Not Executed!!
-----
SELECT IIF (EXISTS (SELECT 1 FROM [order] WHERE orderId=2), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'false' } ],
  output: {},
  rowsAffected: []
}
INSERT INTO [order] (customerId, totalUsd) VALUES (3, 110)
!!Not Executed!!
-----
-----
table 'orderLineItem'
-----
SELECT IIF (EXISTS (SELECT 1 FROM sys.Objects WHERE object_id = object_id(N'orderLineItem') AND type IN (N'U',N'V')), 'true', 'false') as 'object_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { object_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
SELECT IIF (EXISTS (SELECT 1 FROM orderLineItem WHERE orderLineItemId=1), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'false' } ],
  output: {},
  rowsAffected: []
}
INSERT INTO orderLineItem (orderId, productId, productQuantity, productPriceUsd) VALUES (1, 1, 1, 5)
!!Not Executed!!
-----
SELECT IIF (EXISTS (SELECT 1 FROM orderLineItem WHERE orderLineItemId=2), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'false' } ],
  output: {},
  rowsAffected: []
}
INSERT INTO orderLineItem (orderId, productId, productQuantity, productPriceUsd) VALUES (2, 1, 2, 5)
!!Not Executed!!
-----
SELECT IIF (EXISTS (SELECT 1 FROM orderLineItem WHERE orderLineItemId=3), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'false' } ],
  output: {},
  rowsAffected: []
}
INSERT INTO orderLineItem (orderId, productId, productQuantity, productPriceUsd) VALUES (2, 2, 2, 50)
!!Not Executed!!
-----
Executing JSON fixture complete: 00:00:05.8.13.
Closing connection to '(localdb)\node' database 'scratch'...
Connection to '(localdb)\node' database 'scratch' closed: 00:00:00.0.01
Fixture complete: 00:00:05.9.51.
end.
Waiting for the debugger to disconnect...

Process finished with exit code 0

```

**Comment by YardGnomeNinja** [comment]
In `initialize.json` set `disableMutationStatements=false`. Sorry, I added that feature since I last posted. With it set to true it only executes the `EXISTS` queries. The rest are just printed to the output and not executed.

**Comment by TimelordUK** [comment]
OK that is better, now I see the issue - thanks

can you locally replace your reader.js as follows

I am testing this change to check everything still works as expected.

```javascript
/**
 * Created by Stephen on 28/06/2017.
 */

// the main work horse that manages a query from start to finish by interacting with the c++

'use strict'

const readerModule = ((() => {
  function DriverRead (cppDriver, queue) {
    const native = cppDriver
    const workQueue = queue
    let useUTC = true

    const rowBatchSize = 50 /* ignored for prepared statements */
    function setUseUTC (utc) {
      useUTC = utc
    }

    function isInfo (err) {
      return err && err.sqlstate && err.sqlstate.length >= 2 && err.sqlstate.substring(0, 2) === '01'
    }

    /* route non critical info messages to its own event to prevent streams based readers from halting */
    const routeStatementError = (errors, callback, notify) => {
      if (!Array.isArray(errors)) {
        errors = [errors]
      }
      let i = 0
      errors.forEach(err => {
        const info = isInfo(err)
        if (callback && !info) {
          callback(err, null, i < errors.length - 1)
        } else {
          const ev = info ? 'info' : 'error'
          if (notify) {
            notify.emit(ev, err, i < errors.length - 1)
          } else {
            throw new Error(err)
          }
        }
        ++i
      })
    }

    function nativeGetRows (queryId, rowBatchSize) {
      return new Promise((resolve, reject) => {
        native.readColumn(queryId, rowBatchSize, (e, res) => {
          setImmediate(() => {
            if (e) {
              reject(e)
            } else {
              resolve(res)
            }
          })
        })
      })
    }

    // invokeObject.begin(queryId, query, params, onInvoke)

    function Query (notify, queue, query, params, invokeObject, callback) {
      let meta
      let rows = []
      let outputParams = []
      const queryId = notify.getQueryId()
      let queryRowIndex = 0
      let batchRowIndex = 0
      let batchData = null
      let running = true
      let paused = false

      function close () {
        running = false
        workQueue.nextOp()
      }

      function dispatchInfoReturnErrors (e) {
        const infoMessages = []
        const errorMessages = []
        if (e && Array.isArray(e)) {
          e.forEach(errorOrInfo => {
            if (isInfo(errorOrInfo)) {
              infoMessages.push(errorOrInfo)
            } else {
              errorMessages.push(errorOrInfo)
            }
          })
        }
        if (errorMessages.length > 0) {
          return errorMessages
        } else if (infoMessages.length > 0) {
          routeStatementError(infoMessages, callback, notify, false)
        }
        return []
      }

      function nativeNextResult (queryId) {
        return new Promise((resolve, reject) => {
          native.nextResult(queryId, (e, res) => {
            setImmediate(() => {
              // may contain info messages e.g. raised by PRINT statements - do not want to reject these
              const errorMessages = e ? dispatchInfoReturnErrors(e) : []
              if (errorMessages.length > 0) {
                reject(errorMessages)
              }
              resolve(res)
            })
          })
        })
      }

      function beginQuery (queryId) {
        return new Promise((resolve, reject) => {
          invokeObject.begin(queryId, query, params, (e, columnDefinitions, procOutputOrMore) => {
            setImmediate(() => {
              if (e && !procOutputOrMore) {
                reject(e)
              } else {
                resolve({
                  warning: e,
                  columnDefinitions: columnDefinitions,
                  procOutput: procOutputOrMore
                })
              }
            })
          })
        })
      }

      // console.log('fetch ', queryId)
      const dispatchRows = (results) => {
        if (!results) { return }
        if (paused) return
        const resultRows = results.data
        if (!resultRows) { return }
        const numberRows = resultRows.length
        while (batchRowIndex < numberRows) {
          if (paused) {
            break
          }
          const driverRow = resultRows[batchRowIndex]
          notify.emit('row', queryRowIndex)
          batchRowIndex++
          queryRowIndex++
          let currentRow
          if (callback) {
            currentRow = []
            rows[rows.length] = currentRow
          }
          for (let column = 0; column < driverRow.length; ++column) {
            let rowColumn = driverRow[column]
            if (callback) {
              if (rowColumn && useUTC === false) {
                if (meta[column].type === 'date') {
                  rowColumn = new Date(rowColumn.getTime() - rowColumn.getTimezoneOffset() * -60000)
                }
              }
              currentRow[column] = rowColumn
            }
            notify.emit('column', column, rowColumn, false)
          }
        }
      }

      const rowsCompleted = (results, more) => {
        invokeObject.end(queryId, outputParams, (err, r, freeMore, op) => {
          if (callback) {
            callback(err, r, freeMore, op)
          }
          if (!freeMore) {
            setImmediate(() => {
              notify.emit('done')
            })
          }
        }, results, more)
      }

      const rowsAffected = nextResultSetInfo => {
        const rowCount = nextResultSetInfo.rowCount
        const preRowCount = nextResultSetInfo.preRowCount
        const moreResults = !nextResultSetInfo.endOfResults
        notify.emit('rowcount', preRowCount)

        const state = {
          meta: null,
          rowcount: rowCount
        }

        rowsCompleted(state, moreResults)
      }

      const moveToNextResult = (nextResultSetInfo) => {
        setImmediate(() => {
          if (!meta) {
            rowsCompleted(
              {
                meta: meta,
                rows: rows
              },
              !nextResultSetInfo.endOfResults)
          } else if (meta && meta.length === 0) {
            // handle the just finished result reading
            // if there was no metadata, then pass the row count (rows affected)
            rowsAffected(nextResultSetInfo)
          } else {
            rowsCompleted(
              {
                meta: meta,
                rows: rows
              },
              !nextResultSetInfo.endOfResults)
          }

          // reset for the next resultset
          meta = nextResultSetInfo.meta
          if (!meta) {
            nextResult()
            return
          }
          rows = []
          if (nextResultSetInfo.endOfResults) {
            close()
          } else {
            // if this is just a set of rows
            if (meta.length > 0) {
              notify.emit('meta', meta)
              // kick off reading next set of rows
              dispatch()
            } else {
              nextResult()
            }
          }
        })
      }

      function dispatch () {
        if (!running) return
        if (paused) return // will come back at some later stage

        nativeGetRows(queryId, rowBatchSize).then(d => {
          batchRowIndex = 0
          batchData = d
          dispatchRows(d)
          if (!d.end_rows) {
            dispatch()
          } else {
            nextResult()
          }
        }).catch(err => {
          routeStatementError(err, callback, notify, false)
          close()
        })
      }

      const nextResult = () => {
        nativeNextResult(queryId).then(nextResultSetInfo => {
          moveToNextResult(nextResultSetInfo)
        }).catch(err => {
          routeStatementError(err, callback, notify, false)
          close()
        })
      }

      function begin () {
        beginQuery(queryId, query, params).then(res => {
          if (res.warning) {
            routeStatementError(res.warning, callback, notify)
          }
          outputParams = res.outputParams
          meta = res.columnDefinitions
          if (meta.length > 0) {
            notify.emit('meta', meta)
            dispatch()
          } else {
            nextResult()
          }
        }).catch(err => {
          invokeObject.end(queryId, outputParams, () => {
            if (!Array.isArray(err)) {
              err = [err]
            }
            routeStatementError(err, callback, notify)
          }, null, false)
          close()
        })
        notify.emit('submitted', query, params)
      }

      function pause () {
        paused = true
        queue.park(notify.getOperation())
      }

      function resume () {
        queue.resume(notify.getOperation())
        paused = false
        dispatchRows(batchData)
        dispatch()
      }

      return {
        begin: begin,
        pause: pause,
        resume: resume
      }
    }

    function getQuery (notify, query, params, invokeObject, callback) {
      const q = new Query(notify, workQueue, query, params, invokeObject, callback)
      notify.setQueryWorker(q)
      return q
    }

    return {
      setUseUTC: setUseUTC,
      getQuery: getQuery
    }
  }

  return {
    DriverRead: DriverRead
  }
})())

exports.readerModule = readerModule

```

**Comment by TimelordUK** [comment]
I should add here is my output with this change

```console
"C:\Program Files\nodejs\node.exe" C:\Users\Stephen\dev\js\json-to-sql\dist\app.js ../test/fixtures/initialize.json
Opening connection to '(localdb)\node' database 'scratch'...
Opening connection complete: 00:00:00.0.41.
Executing JSON fixture '../test/fixtures/initialize.json'...
-----
table 'customer'
-----
SELECT IIF (EXISTS (SELECT 1 FROM sys.Objects WHERE object_id = object_id(N'customer') AND type IN (N'U',N'V')), 'true', 'false') as 'object_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { object_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
SELECT IIF (EXISTS (SELECT 1 FROM customer WHERE customerId=1), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
UPDATE customer SET first='Mike', last='Wazowski' WHERE customerId=1
{
  recordsets: [],
  recordset: undefined,
  output: {},
  rowsAffected: [ 1 ]
}
-----
SELECT IIF (EXISTS (SELECT 1 FROM customer WHERE customerId=2), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
UPDATE customer SET first='James', last='Sullivan' WHERE customerId=2
{
  recordsets: [],
  recordset: undefined,
  output: {},
  rowsAffected: [ 1 ]
}
-----
SELECT IIF (EXISTS (SELECT 1 FROM customer WHERE customerId=3), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
UPDATE customer SET first='Boo', last='The Kid' WHERE customerId=3
{
  recordsets: [],
  recordset: undefined,
  output: {},
  rowsAffected: [ 1 ]
}
-----
-----
table 'product'
-----
SELECT IIF (EXISTS (SELECT 1 FROM sys.Objects WHERE object_id = object_id(N'product') AND type IN (N'U',N'V')), 'true', 'false') as 'object_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { object_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
SELECT IIF (EXISTS (SELECT 1 FROM product WHERE productId=1), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
UPDATE product SET name='Spatula', description='flip flip flip', priceUsd=5 WHERE productId=1
{
  recordsets: [],
  recordset: undefined,
  output: {},
  rowsAffected: [ 1 ]
}
-----
SELECT IIF (EXISTS (SELECT 1 FROM product WHERE productId=2), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
UPDATE product SET name='Hat', description=null, priceUsd=50 WHERE productId=2
{
  recordsets: [],
  recordset: undefined,
  output: {},
  rowsAffected: [ 1 ]
}
-----
-----
table '[order]'
-----
SELECT IIF (EXISTS (SELECT 1 FROM sys.Objects WHERE object_id = object_id(N'[order]') AND type IN (N'U',N'V')), 'true', 'false') as 'object_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { object_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
SELECT IIF (EXISTS (SELECT 1 FROM [order] WHERE orderId=1), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
UPDATE [order] SET customerId=1, totalUsd=15, createdOn='2020-1-1 14:00:00' WHERE orderId=1
{
  recordsets: [],
  recordset: undefined,
  output: {},
  rowsAffected: [ 1 ]
}
-----
SELECT IIF (EXISTS (SELECT 1 FROM [order] WHERE orderId=2), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
UPDATE [order] SET customerId=3, totalUsd=110 WHERE orderId=2
{
  recordsets: [],
  recordset: undefined,
  output: {},
  rowsAffected: [ 1 ]
}
-----
-----
table 'orderLineItem'
-----
SELECT IIF (EXISTS (SELECT 1 FROM sys.Objects WHERE object_id = object_id(N'orderLineItem') AND type IN (N'U',N'V')), 'true', 'false') as 'object_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { object_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
SELECT IIF (EXISTS (SELECT 1 FROM orderLineItem WHERE orderLineItemId=1), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
UPDATE orderLineItem SET orderId=1, productId=1, productQuantity=1, productPriceUsd=5 WHERE orderLineItemId=1
{
  recordsets: [],
  recordset: undefined,
  output: {},
  rowsAffected: [ 1 ]
}
-----
SELECT IIF (EXISTS (SELECT 1 FROM orderLineItem WHERE orderLineItemId=2), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
UPDATE orderLineItem SET orderId=2, productId=1, productQuantity=2, productPriceUsd=5 WHERE orderLineItemId=2
{
  recordsets: [],
  recordset: undefined,
  output: {},
  rowsAffected: [ 1 ]
}
-----
SELECT IIF (EXISTS (SELECT 1 FROM orderLineItem WHERE orderLineItemId=3), 'true', 'false') as 'record_exists'
{
  recordsets: [ [ [Object] ] ],
  recordset: [ { record_exists: 'true' } ],
  output: {},
  rowsAffected: []
}
UPDATE orderLineItem SET orderId=2, productId=2, productQuantity=2, productPriceUsd=50 WHERE orderLineItemId=3
{
  recordsets: [],
  recordset: undefined,
  output: {},
  rowsAffected: [ 1 ]
}
-----
Executing JSON fixture complete: 00:00:00.0.87.
Closing connection to '(localdb)\node' database 'scratch'...
Connection to '(localdb)\node' database 'scratch' closed: 00:00:00.0
Fixture complete: 00:00:00.1.9.
end.

Process finished with exit code 0

```

**Comment by YardGnomeNinja** [comment]
Looks like that does the trick! Thanks so much for looking into it!
